### ISSUE TITLE
custodia doesn't  handle SSLError Exceptions 

### ISSUE THREAD

**Issue by mrniranjan** [latchset/custodia#200]
In httpd/server.py under class HTTPServer, there are no exception handle when SSL errors occur,  Can we have SSLError exception handling , specifically for ssl.create_default_context() and  context.load_cert_chain()

**Comment by raildo** [comment]
Sure

**Comment by mrniranjan** [comment]
I am hoping by handling the SSLError exception, a meaningful message will be displayed to the user in case the certificates (including keys) are not of proper order/invalid file/file doesn't exist/invalid chain, not just raise the SSLError.

**Comment by mrniranjan** [comment]
This patch fails because the ForkingTLSServer class doesn't have an attribute logger. If the patch want to use the global variable logger,  it should be just logger.debug(msg) not self.logger.debug(msg),

**Comment by mrniranjan** [comment]
Also the patch has another issue:
msg = ("Error during TLS Server creation, invalid parameters. "
+                   "Certificate Authority file: %s, Certificate Path: "
+                   "%, Key file: %s and/or Certificate "
+                   "file: %s. Please reverify this data and try "
+                   "again", cafile, capath, keyfile, certfile)

The above should be instead

msg = ("Error during TLS Server creation, invalid parameters. "
+                   "Certificate Authority file: %s, Certificate Path: "
+                   "%s, Key file: %s and/or Certificate "
+                   "file: %s. Please reverify this data and try "
+                   "again"  %( cafile, capath, keyfile, certfile))

or 

msg = ("Error during TLS Server creation, invalid parameters. "
+                   "Certificate Authority file: %s, Certificate Path: "
+                   "%s, Key file: %s and/or Certificate "
+                   "file: %s. Please reverify this data and try "
+                   "again"  %( str(cafile), str(capath), str(keyfile), str(certfile)))

**Comment by tiran** [comment]
I have implemented an alternative approach in PR #210. Instead of creating a new exception, the code now logs parameters for each step. This makes it easier to locate ``No such file or directory`` problems. You have to look for one or two files instead of four.

```
2017-06-29 13:06:59 - custodia                         - Custodia debug logger enabled
2017-06-29 13:06:59 - custodia                         - Custodia audit log: /home/heimes/dev/redhat/custodia/custodia.audit.log
2017-06-29 13:06:59 - custodia                         - Custodia instance <main>
2017-06-29 13:06:59 - custodia                         - Config file(s) ['/home/heimes/dev/redhat/custodia/custodia.conf'] loaded
2017-06-29 13:06:59 - server                           - Serving on 0.0.0.0:10443 (HTTPS)
2017-06-29 13:06:59 - server                           - Creating SSLContext for TLS server (cafile: 'tests/ca/custodia-ca.pem', capath: 'None', verify client: True).
2017-06-29 13:06:59 - server                           - Loading cert chain 'tests/ca/custodia-server.pem' (keyfile: 'tests/ca/custodia-server.key')
2017-06-29 13:06:59 - server                           - Failed to create a SSLContext for TLS server: [Errno 2] No such file or directory
Traceback (most recent call last):
  File "/home/heimes/dev/redhat/custodia/bin/custodia", line 25, in <module>
    main()
  File "/home/heimes/dev/redhat/custodia/bin/custodia", line 22, in main
    sys.exit(func())
  File "/home/heimes/dev/redhat/custodia/src/custodia/server/__init__.py", line 140, in main
    httpd = HTTPServer(config['server_url'], config)
  File "/home/heimes/dev/redhat/custodia/src/custodia/httpd/server.py", line 501, in __init__
    self.httpd = serverclass(address, self.handler, config)
  File "/home/heimes/dev/redhat/custodia/src/custodia/httpd/server.py", line 128, in __init__
    self._context = self._mkcontext()
  File "/home/heimes/dev/redhat/custodia/src/custodia/httpd/server.py", line 162, in _mkcontext
    context.load_cert_chain(certfile, keyfile)
FileNotFoundError: [Errno 2] No such file or directory
Makefile:129: recipe for target 'run' failed
make: *** [run] Error 1
```
